src/kzg.rs

```diff
@@ -0,0 +1,144 @@
+use crate::bytes::write_bytes_to_lower_hex;
+use crate::primitives::Bytes32;
+use crate::serde::{try_bytes_from_hex_str, HexError};
+use crate::ssz::ByteVector;
+use core::fmt;
+use ssz_rs::prelude::*;
+use std::ops::{Deref, DerefMut};
+use thiserror::Error;
+
+const KZG_COMMITMENT_BYTES_LEN: usize = 48;
+const KZG_PROOF_BYTES_LEN: usize = 48;
+
+pub type VersionedHash = Bytes32;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[cfg(feature = "serde")]
+    #[error("error deserializing hex-encoded input: {0}")]
+    Hex(#[from] HexError),
+    #[error(
+        "invalid length of encoding: expected {expected} bytes but only provided {provided} bytes"
+    )]
+    EncodingError { provided: usize, expected: usize },
+}
+
+#[derive(Default, Debug, Clone, SimpleSerialize, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(into = "String", try_from = "String"))]
+pub struct KzgCommitment(ByteVector<KZG_COMMITMENT_BYTES_LEN>);
```

Comment:
I think for now we can just get away with `pub type KzgCommitment = ByteVector<KZG_COMMITmENT_BYTES_LEN>;`

and drop the rest of the code defining/impl'ing `KzgCommitment`

Reply:
👍